### DISCUSSION TITLE
More dynamic setting of attributes

### DISCUSSION THREAD

**Discussion by rmccu** [jongpie/NebulaLogger#381] [Ideas]
https://github.com/jongpie/NebulaLogger/blob/3b04138cdb22211796870ef49d8b3773465641e9/nebula-logger/core/main/log-management/classes/LogEntryEventHandler.cls#L162

Rather than have these massive blocks of code that are basically copying the attributes, it seems like we could read the fields from schema and then loop through them. In the for loop you could have a check to process specific fields in special ways, otherwise just copy the data over referencing the fields using get/set methods. This would remove a lot of lines of code, and make this more resilient so that if you added a new field in the future you wouldn't need to update this code unless you wanted to handle it special.

**Comment by jongpie** [comment]
@rmccu I prefer the massive blocks of code, instead of trying to dynamically loop through the schema. The code blocks are fairly easy to read (since each line is just mapping 1 field from `LogEntryEvent__e` to an equivalent field on `Log__c` or `LogEntry__c`), and provide some benefits:

- Mapping the fields in code explicitly shows the intent of which fields on `LogEntryEvent__e` should map to fields on either `Log__c` or `LogEntry__c` - dynamically mapping fields is a little bit more cryptic/less intuitive
- Mapping the fields in code provides compile-time validation errors, which helps tremendously with development & testing
- Not all fields on `LogEntryEvent__e` have the same name on `Log__c` or `LogEntry__c` (e.g., `LogEntryEvent__e.LoggedById__c` maps to `Log__c.LoggedBy__c`), so dynamically mapping wouldn't always work
- Not all fields on `LogEntryEvent__e` have the same data type as the mapped fields `Log__c` or `LogEntry__c`, so dynamically mapping wouldn't always work

**Reply by jamessimone** [reply]
Another bullet point in favor of explicit setting here — it’s faster. In this case, the logic is being run async so there’s less of an immediate need for every single millisecond to be finely optimized, but as a counterpoint since you can have your own handlers defined for Log Entry Events that run after this handler, avoiding the (slower) speed of Schema looping is nice.
